### PR TITLE
feat(components): Handle Stopped state for all components and fix bug in robomaker simulation function

### PR DESCRIPTION
**Pull request by akartsky**
Mark component as failed if they reach stopped state 
and 
Fix bug in robomaker simulation function

https://github.com/kubeflow/pipelines/issues/6465

### CODE REVIEW

File: components/aws/sagemaker/train/src/sagemaker_training_component.py [hunk 1]
```diff
@@ -64,6 +64,14 @@ def _get_job_status(self) -> SageMakerJobStatus:
                 error_message=message,
                 raw_status=status,
             )
+        if status == "Stopped":
+            message = "The job reached Stopped state"
+            return SageMakerJobStatus(
+                is_completed=True,
+                has_error=True,
+                error_message=message,
```
Comment (surajkota) [inline components/aws/sagemaker/train/src/sagemaker_training_component.py:72]: is the error message always present for stopped jobs? Does this code work fine even if the message is not present?

**Comment by surajkota** [comment]
Do you mind pulling changes from this PR as well? - https://github.com/kubeflow/pipelines/pull/5993 so we can update the container once

File: components/aws/sagemaker/train/src/sagemaker_training_component.py (hunk 1 above)
Comment (akartsky) [reply components/aws/sagemaker/train/src/sagemaker_training_component.py:72]: i manually add the `message` on line 68
ya `error_message` in optional

**Comment by akartsky** [comment]
merged in this PR : https://github.com/kubeflow/pipelines/pull/8336

**Pull request closed without merging**